pkg/rm/rm_remoting_test.go

```diff
@@ -0,0 +1,190 @@
+package rm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/seata/seata-go/pkg/common"
+	"github.com/seata/seata-go/pkg/protocol/branch"
+	"github.com/seata/seata-go/pkg/tm"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testResourceManager     *TestResourceManager
+	onceTestResourceManager = &sync.Once{}
+)
+
+func TestGetRMRemotingInstance(t *testing.T) {
+	tests := struct {
+		name string
+		want *RMRemoting
+	}{"test1", &RMRemoting{}}
+
+	t.Run(tests.name, func(t *testing.T) {
+		assert.Equalf(t, tests.want, GetRMRemotingInstance(), "GetRMRemotingInstance()")
+	})
+
+}
+
+func TestGetRmCacheInstance(t *testing.T) {
+
+	tests := struct {
+		name string
+		want *ResourceManagerCache
+	}{"test1", &ResourceManagerCache{}}
+
+	t.Run(tests.name, func(t *testing.T) {
+		GetRmCacheInstance().RegisterResourceManager(GetTestResourceManagerInstance())
+		actual := GetRmCacheInstance().GetResourceManager(branch.BranchTypeTCC)
+		assert.Equalf(t, GetTestResourceManagerInstance(), actual, "GetRmCacheInstance()")
+	})
+
+}
+
+type TestResource struct {
+	ResourceGroupId string `default:"DEFAULT"`
+	AppName         string
+	*TwoPhaseAction
+}
+
+func (t *TestResource) GetResourceGroupId() string {
```

Comment:
i think test code do not nedd to export
```suggestion
func (t *TestResource) getResourceGroupId() string {
```

Reply:
The capitalization here is not an export, it is an implementation interface